coderd/provisionerjobs_test.go

```diff
@@ -27,162 +27,255 @@ import (
+			ctx := testutil.Context(t, testutil.WaitMedium)
+			jobs, err := templateAdminClient.OrganizationProvisionerJobs(ctx, owner.OrganizationID, &codersdk.OrganizationProvisionerJobsOptions{
+				Limit: 1,
+			})
+			require.NoError(t, err)
+			require.Len(t, jobs, 1)
+		})
+
+		// For now, this is not allowed even though the member has created a
+		// workspace. Once member-level permissions for jobs are supported
+		// by RBAC, this test should be updated.
+		t.Run("MemberDenied", func(t *testing.T) {
+			t.Parallel()
+			ctx := testutil.Context(t, testutil.WaitMedium)
+			jobs, err := memberClient.OrganizationProvisionerJobs(ctx, owner.OrganizationID, nil)
+			require.Error(t, err)
+			require.Len(t, jobs, 0)
 		})
-		require.NoError(t, err)
-		require.Len(t, jobs, 1)
 	})
 
-	t.Run("Limit", func(t *testing.T) {
+	// Ensures that when a provisioner job is in the succeeded state,
+	// the API response includes both worker_id and worker_name fields
+	t.Run("AssignedProvisionerJob_IncludesWorkerIDAndName", func(t *testing.T) {
 		t.Parallel()
-		ctx := testutil.Context(t, testutil.WaitMedium)
-		jobs, err := templateAdminClient.OrganizationProvisionerJobs(ctx, owner.OrganizationID, &codersdk.OrganizationProvisionerJobsOptions{
-			Limit: 1,
+
+		db, ps := dbtestutil.NewDB(t, dbtestutil.WithDumpOnFailure())
+		client, _, coderdAPI := coderdtest.NewWithAPI(t, &coderdtest.Options{
+			IncludeProvisionerDaemon: false,
+			Database:                 db,
+			Pubsub:                   ps,
 		})
+		provisionerDaemonName := "provisioner_daemon_test"
+		provisionerDaemon := coderdtest.NewTaggedProvisionerDaemon(t, coderdAPI, provisionerDaemonName, map[string]string{"owner": "", "scope": "organization"})
+		owner := coderdtest.CreateFirstUser(t, client)
+		templateAdminClient, _ := coderdtest.CreateAnotherUser(t, client, owner.OrganizationID, rbac.ScopedRoleOrgTemplateAdmin(owner.OrganizationID))
+
+		version := coderdtest.CreateTemplateVersion(t, client, owner.OrganizationID, nil)
+		coderdtest.AwaitTemplateVersionJobCompleted(t, client, version.ID)
+		template := coderdtest.CreateTemplate(t, client, owner.OrganizationID, version.ID)
+
+		workspace := coderdtest.CreateWorkspace(t, client, template.ID)
+		coderdtest.AwaitWorkspaceBuildJobCompleted(t, client, workspace.LatestBuild.ID)
+
+		// Stop the provisioner so it doesn't grab any more jobs
+		err := provisionerDaemon.Close()
 		require.NoError(t, err)
-		require.Len(t, jobs, 1)
-	})
 
-	// For now, this is not allowed even though the member has created a
-	// workspace. Once member-level permissions for jobs are supported
-	// by RBAC, this test should be updated.
-	t.Run("MemberDenied", func(t *testing.T) {
-		t.Parallel()
-		ctx := testutil.Context(t, testutil.WaitMedium)
-		jobs, err := memberClient.OrganizationProvisionerJobs(ctx, owner.OrganizationID, nil)
-		require.Error(t, err)
-		require.Len(t, jobs, 0)
+		t.Run("List_AssignedProvisionerJob_IncludesWorkerIDAndName", func(t *testing.T) {
```

Comment:
```suggestion
		t.Run("List_IncludesWorkerIDAndName", func(t *testing.T) {
```
With the above suggestion, this would result in `AssignedProvisionerJob/List_IncludesWorkerIDAndName` removing duplication. (Same can be applied to test below.)